### ISSUE TITLE
simpler syntax for csvread

### ISSUE THREAD

**Issue by danielkorzekwa** [scalanlp/breeze#441]
csvread("src/test/resources/cogp/cogp_no_missing_points.csv")

in addition to existing one:
csvread(new File("src/test/resources/cogp/cogp_no_missing_points.csv"))

**Comment by DonBeo** [comment]
I'm relatively new to scala but I would like to try to solve this issue. 

I think a possible solution would be something similar to the accepted answer to this question: http://stackoverflow.com/questions/3508077/how-to-define-type-disjunction-union-types 

the `file` variable should then be `string` or `java.io.File`.    
What do you think?

**Comment by dlwh** [comment]
I honestly don't like using Strings as arguments to methods like that, because it's not clear if the string is going to be treated as a file, or parsed directly.

I would be onboard with an implicit that added a method like .toFile and/or file"..." for string literals:

`csvread(file"src/test/resources/cogp/cogp_no_missing_points.csv")` doesn't seem so onerous?

**Comment by DonBeo** [comment]
I think this solution is better. I discovered that it is quiet complex to override a function with default values.  If this is fine for you I'll implement this change in the following days.

**Comment by dlwh** [comment]
sounds great!

**Comment by danielkorzekwa** [comment]
From the user point of view it would be much better to call
```csvread('somefile.csv')``` instead of ```csvread(file'somefile.csv') ```

Sounds like a little difference but when you call it many many times then it makes the difference for the user. Currently, to read csv file in both R and Scala:

R:
```
 read.csv('somefile.csv')
```

Scala:
```
import breeze.linalg._
import java.io.File
csvread(new File('somefile.csv'),skiplines=1)
```

**Comment by dlwh** [comment]
I dunno, I'm inclined to stick to my guns on this one. I don't like the ambiguity between parsing a string and reading a file at a path. We have strong static types for a reason. (Maybe an additional "terse mode" import to allow what you want?)

**Comment by DonBeo** [comment]
I agree with David.  In addition is quiet complex to override a function in order to make it able to accept 2 different types. 
An alternative would be to define a new function: `readlocalcsv` that takes as input a string.

**Comment by frgomes** [comment]
IMHO, ``csvread(reader: java.io.Reader,...`` is preferable, since ``csvread`` could participate in a pipeline. A similar concept would be ``csvwrite(writer: java.io.Writer,...``, which can offload data to the next participant of a pipeline.

In my branch I've created ``csvreader`` and ``csvwriter`` as below, in order to avoid collision with ``csvread`` and ``csvwrite``:

```scala
  def csvreader(
               reader: java.io.Reader,
               separator: Char = ',',
               quote: Char = '"',
               escape: Char = '\\',
               skipLines: Int = 0): DenseMatrix[Double] = {
    var mat = CSVReader.read(reader, separator, quote, escape, skipLines)
    mat = mat.takeWhile(line => line.length != 0 && line.head.nonEmpty) // empty lines at the end
    reader.close()
    if (mat.length == 0) {
      DenseMatrix.zeros[Double](0, 0)
    } else {
      DenseMatrix.tabulate(mat.length, mat.head.length)((i, j) => mat(i)(j).toDouble)
    }
  }

  def csvwriter(
                writer: java.io.Writer,
                mat: Matrix[Double],
                separator: Char = ',',
                quote: Char = '\u0000',
                escape: Char = '\\',
                skipLines: Int = 0): Unit = {
    CSVWriter.write(writer, IndexedSeq.tabulate(mat.rows, mat.cols)(mat(_, _).toString), separator, quote, escape)
  }
```

Then you can do even small test cases playing with data in memory, such as:
```scala
val inventory =
  Seq(
    "Item,Qty,Price",
    "Banana,10,0.52",
    "Apple",24,0.60",
    "Apple Computer",1,4560.00")
val text  = inventory.mkString("\n")
val bytes = text.getBytes(java.nio.charset.StandardCharsets.UTF_8.name)
val is: java.io.InputStream = new java.io.ByteArrayInputStream(bytes)
val reader = InputStreamReader(is)
csvreader(reader, ...)
```
